erpnext/hr/doctype/compensatory_leave_request/compensatory_leave_request.py

```diff
@@ -4,9 +4,127 @@
 
 from __future__ import unicode_literals
 import frappe
+from frappe import _
+from frappe.utils import flt, formatdate, getdate
 from frappe.model.document import Document
+from erpnext.hr.doctype.leave_allocation.leave_allocation import get_carry_forwarded_leaves
 
 class CompensatoryLeaveRequest(Document):
 	def validate_present(self):
 		pass
 
+	def validate(self):
+		self.validate_dates()
+		self.validate_overlap()
+
+	def on_submit(self):
+		if not self.leave_type:
+			frappe.throw(_("Please select a leave type to submit the request"))
+		else:
+			company = frappe.db.get_value("Employee", self.employee, "company")
+			leave_period = self.get_leave_period(company)
+			if leave_period:
+				max_leaves_allowed = frappe.db.get_value("Leave Type", self.leave_type, "max_leaves_allowed")
+				self.validate_leave_allocation_days(leave_period, max_leaves_allowed)
+				self.create_leave_allocation(leave_period, max_leaves_allowed)
+
+	def create_leave_allocation(self, leave_period, max_leaves_allowed):
+		leave_allocated = get_leave_allocation_for_period(self.employee, self.leave_type, leave_period[0].from_date, leave_period[0].to_date)
+		leave_alloc_balance = max_leaves_allowed - leave_allocated
```

Comment:
We should not take decision here and deduct leaves.
Let's give a validation message saying you can only take that much comp leaves.